core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/util/HttpChecksumUtils.java

```diff
@@ -127,4 +140,36 @@ private static boolean isTrailerChecksumPresent(SdkHttpRequest sdkHttpRequest, C
         }
         return false;
     }
+
+    /**
+     * Calculates the content length of a request. If the content-length isn't in the header,
+     * the method reads the whole input stream to get the length.
+     */
+    public static <T> long calculateContentLength(T payload) {
```

Comment:
why is this method in HttpChecksumUtils when it doesn't have to directly do with Checksums? Probably SignerUtils?

Reply:
Sure.